@jbrowse/react-linear-genome-view fail to compile in Next.js since release 1.0.4

I have tried to update @jbrowse/react-linear-genome-view to 1.0.4 today, but since, I can't compile jbrowse anymore and I get this error:

```
./node_modules/@gmod/indexedfasta/dist/localFile.js:29:0
Module not found: Can't resolve 'fs'
null
```

Also, after reversing to 1.0.3 the error stay, so it is certainly due to a change in one of the dependencies ?

**To Reproduce**

Try to open a page with @jbrowse/react-linear-genome-view.

**Version:**

@jbrowse/react-linear-genome-view: 1.0.4 (but error still here after a reverse to 1.0.3)
next: 10.0.5
react: 16.13.1

Hi @garrettjstevens, hi @elliothershberg, thank for your answers.

My version of @gmod/indexedfasta is `1.1.0`.
I've tried with to use the version `1.0.13` but after an `yarn upgrade @gmod/indexedfasta@1.0.13`,
`yarn info @gmod/indexedfasta version` tell me it's still using the `1.1.0` version.

I've seen that all the dependencies are in `1.0.4`  (like @jbrowse/core for example). Are you aware of a way to pin all of the deps from @jbrowse in `1.0.3` easily  ?

@elliothershberg, unfortunately, the problem is happening at the build phase, not in the browser, as you can see from my NextJs docker : 

```
web_1     | event - build page: /genomeBrowser
web_1     | wait  - compiling...
web_1     | error - ./node_modules/@jbrowse/plugin-sequence/node_modules/@gmod/indexedfasta/dist/localFile.js:29:0
web_1     | Module not found: Can't resolve 'fs'
web_1     | null
```

--- a/products/jbrowse-react-linear-genome-view/.storybook/preview.js
+++ b/products/jbrowse-react-linear-genome-view/.storybook/preview.js
@@ -7,7 +7,7 @@ export const parameters = {
         'Default Sessions',
         'Linear View',
         'Nextstrain View',
-        'Next.js Usage'
+        'Next.js Usage',
       ],
       locales: '',
     },

--- a/products/jbrowse-react-linear-genome-view/.storybook/preview.js
+++ b/products/jbrowse-react-linear-genome-view/.storybook/preview.js
@@ -7,6 +7,7 @@ export const parameters = {
         'Default Sessions',
         'Linear View',
         'Nextstrain View',
+        'Next.js Usage'
       ],
       locales: '',
     },

--- a/products/jbrowse-react-linear-genome-view/README.md
+++ b/products/jbrowse-react-linear-genome-view/README.md
@@ -55,7 +55,7 @@ $ npm install @jbrowse/react-linear-genome-view
 
 ## Documentation
 
-The latest Storybook documentation for the component is hosted [here](https://jbrowse.org/storybook/lgv/master).
+The latest Storybook documentation for the component is hosted [here](https://jbrowse.org/storybook/lgv/main).
 
 ### Note on fonts
 

--- a/products/jbrowse-react-linear-genome-view/README.md
+++ b/products/jbrowse-react-linear-genome-view/README.md
@@ -53,6 +53,10 @@ Or with [npm](https://npmjs.org/):
 $ npm install @jbrowse/react-linear-genome-view
 ```
 
+## Documentation
+
+The latest Storybook documentation for the component is hosted [here](https://jbrowse.org/storybook/lgv/master).
+
 ### Note on fonts
 
 [Roboto](https://fonts.google.com/specimen/Roboto) is the recommended font for

--- a/products/jbrowse-react-linear-genome-view/stories/NextJsUsage.stories.mdx
+++ b/products/jbrowse-react-linear-genome-view/stories/NextJsUsage.stories.mdx
@@ -1,0 +1,103 @@
+import { Meta } from '@storybook/addon-docs/blocks'
+
+<Meta title="Next.js Usage" />
+
+# Usage with Next.js
+
+[Next.js](https://nextjs.org/) is a popular framework for building
+server-rendered React websites with Node.js.
+Given that the JB2 LGV React component is designed for use in static
+apps, that are a few extra steps that need to be taken when using
+the component in a Next.js app.
+
+## Configuration
+
+It is important to understand that in Next.js apps, webpack compilation
+happens twice: once for the server and once for the client.
+Because of this, the following additional webpack configuration needs
+to be added to your `next.config.js` file:
+
+```js
+module.exports = {
+  webpack: (config, { isServer }) => {
+    // Fixes npm packages that depend on `fs` module
+    if (!isServer) {
+      config.node = {
+        fs: 'empty',
+      }
+    }
+
+    return config
+  },
+}
+```
+
+This resolves issues with how our data parsers work, since they are written
+to work in either Node.js or in the browser.
+
+## Dynamic imports
+
+Some React components, such as the React LGV component are not written to
+be compatible with server-side rendering, since they rely on APIs that exist
+only in the browser.
+For these types of components, it is necessary to use [dynamic imports](https://nextjs.org/docs/advanced-features/dynamic-import).
+
+Here is an example of how to dynamically load a component using the JB2 component into a Next.js page:
+
+```js
+// In components/Browser.js
+
+import {
+  createViewState,
+  createJBrowseTheme,
+  JBrowseLinearGenomeView,
+  ThemeProvider,
+} from '@jbrowse/react-linear-genome-view'
+
+const theme = createJBrowseTheme()
+
+const assembly = {
+  // configuration
+}
+
+const tracks = [
+  // configuration
+]
+
+const defaultSession = {
+  // configuration
+}
+
+function View() {
+  const state = createViewState({
+    assembly,
+    tracks,
+    location: '10:29,838,737..29,838,819',
+    defaultSession,
+  })
+  return (
+    <ThemeProvider theme={theme}>
+      <JBrowseLinearGenomeView viewState={state} />
+    </ThemeProvider>
+  )
+}
+
+export default View
+```
+
+Then, in the page where the component is used:
+
+```js
+// in pages/index.js
+import dynamic from 'next/dynamic'
+
+const Browser = dynamic(() => import('../components/Browser'), { ssr: false })
+
+export default function Home() {
+  return (
+    <div>
+      <Browser />
+    </div>
+  )
+}
+```

